src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/Contract.cs

```diff
@@ -7,20 +7,40 @@
 using System;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
 
 namespace Roslyn.Utilities
 {
     internal static class Contract
     {
+        public static Exception Unreachable
+            => ExceptionUtilities.Unreachable;
```

Comment:
Can we dedup these? Or what benefit do we get via having Contract.Unreachable as well?

Reply:
Yes, I'd like to. It will be a massive change though, so I think we should do it in separate PR.

Reply:
It will also affect compiler since ExceptionUtilities are shared and compiler doesn't have Contract.
We can perhaps keep ExceptionUtilities in the compiler in phase one and remove them from the IDE.

Reply:
But why not just use ExceptionUtiltiies.Unreachable then? Are we trying to split up and just have two seprate set of helpers? I guess I'm still not sure what the goal is here.

Reply:
Will use ExceptionUtilities directly.